reporters/kamon-opentelemetry/src/main/scala/kamon/otel/SpanConverter.scala

```diff
@@ -0,0 +1,211 @@
+
+  /** Creates a AnyValue type KeyValue */
+  private[otel] def anyKeyValue(key:String, value:AnyValue):KeyValue = KeyValue.newBuilder().setKey(key.replaceAll("-", ".")).setValue(value).build
+  /** Creates a String type KeyValue */
+  private[otel] def stringKeyValue(key:String, value:String):KeyValue = anyKeyValue(key, AnyValue.newBuilder().setStringValue(value).build())
+  /** Creates a Boolean type KeyValue */
+  private[otel] def booleanKeyValue(key:String, value:Boolean):KeyValue = anyKeyValue(key, AnyValue.newBuilder().setBoolValue(value).build())
+  /** Creates a Long type KeyValue */
+  private[otel] def longKeyValue(key:String, value:Long):KeyValue = anyKeyValue(key,AnyValue.newBuilder().setIntValue(value).build())
+
+  /**
+   * Converts a sequence of Kamon spans to a proto ''ResourceSpans''
+   * @param resource resource information for this instance of the service. Added as resource labels
+   * @param instrumentationLibrary instrumentation library information to add to the exported spans
+   * @param spans The spans to export
+   * @return
+   */
+  private[otel] def toProtoResourceSpan(resource:Resource, instrumentationLibrary:InstrumentationLibrary)(spans:Seq[Span.Finished]):ResourceSpans = {
+    import collection.JavaConverters._
+
+    //it is assumed all spans belong to the same instrumentation library
+    val instrumentationLibrarySpans = InstrumentationLibrarySpans.newBuilder()
+      .setInstrumentationLibrary(instrumentationLibrary)
+      .addAllSpans(spans.map(toProtoSpan).asJava)
+      .build()
+
+    ResourceSpans.newBuilder()
+      .setResource(resource)
+      .addAllInstrumentationLibrarySpans(Collections.singletonList(instrumentationLibrarySpans))
+      .build()
+  }
+
+  /**
+   * Converts a Kamon span to a proto span
+   * @param span the span to convert
+   * @return
+   */
+  private[otel] def toProtoSpan(span:Span.Finished):ProtoSpan = {
+    //converts Kamon span tags to proto KeyValue attributes
+    val attributes:List[KeyValue] = span.tags.iterator.map(toProtoKeyValue).toList
+
+    //converts Kamon span links to proto links
+    val links:Seq[ProtoSpan.Link] = span.links.map(toProtoLink)
+
+    import collection.JavaConverters._
+    val builder = ProtoSpan.newBuilder()
+      .setTraceId(traceIdToByteString(span.trace.id))
+      .setSpanId(spanIdToByteString(span.id))
+      .setName(span.operationName)
+      .setKind(toProtoKind(span.kind))
+      .setStartTimeUnixNano(toEpocNano(span.from))
+      .setEndTimeUnixNano(toEpocNano(span.to))
+      .setStatus(getStatus(span))
+      .addAllAttributes(attributes.asJava)
+      .addAllLinks(links.asJava)
+
+    //TODO add set traceState once we have something to set. Need w3c context
+    //It is a trace_state in w3c-trace-context format: https://www.w3.org/TR/trace-context/#tracestate-header
+    //See also https://github.com/w3c/distributed-tracing for more details about this field.
+    //builder.setTraceState()
```

Comment:
Can this be done with the current support for w3c trace context?

Reply:
Unsure where to get it from, the TraceState is stored in the Context
`context.withEntry(Span.Key, span).withEntry(TraceStateKey, traceState)`
https://github.com/kamon-io/Kamon/blob/master/core/kamon-core/src/main/scala/kamon/trace/SpanPropagation.scala#L56

But as the reporter only sees the `Span.Finished` which does not contain any W3C specific data.   
IMO one would need to update the `Span.Finished` case class to contain some meta fields or similar containing the `TraceState` key/value

Reply:
We'll leave it for a later update then!